.github/workflows/toolchain-upgrade.yml

```diff
@@ -0,0 +1,34 @@
+# Copyright Kani Contributors
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+
+# We use block scalar notation to allow us to add ":" to the workflow name.
+name: >-
+  Every 4 hours: Attempt toolchain update
+
+on:
+  schedule:
+    - cron: "*/4 42 * * *" # Run this every four hours at minute 42
```

Comment:
Why minute 42?

Reply:
I have added a further comment in #2319: this is just a random choice to avoid the typical peak times of 0 or 30 minutes past the hour..